encodings/BinaryEncoding.md

```diff
@@ -10,37 +14,45 @@ This will allow us to, in 1 deprecation cycle to completely switch to a new form
 ## Version Format (version_id = 0)
 The version format for the version_id = 0 is based on ideas from proto encoding. The main 
 requirements are to allow adding and removing fields in less than 1 deprecation cycle. It
-contains a list of repeated fields:
+contains a list of fields:
 
 `<field><field>...`
 
 ### Field
+Each field is a 1-byte field ID paired with a field value, where the format of the field value is
+determined by both the field ID and the data type. For example, field 0 in Trace Context may
+have a completely different format than field 0 in Tag Context or field 1 in Trace Context.
+
 Each field that we send on the wire will have the following format:
 
 `<field_id><field_format>`
 
 * `field_id` is a single byte.
 
-* `field_format` must be defined for each metadata field separately, that means that for field_id
- = 0 in trace context the field_value may have a completely different representation than the 
- field_id = 0 in the server-stats metadata.
+* `field_format` must be defined for each field separately.
+
+The specification for a data type's format must also specify whether each field is single or
+repeated. For example, String tag in Tag Context is repeated, and Trace-id in Trace Context in
+single. Every single field is optional. The specification for a data type's format MAY define a
```

Comment:
> Every single field is optional.

I feel a little ambiguous here. Does it mean "all the fields are optional" or "only single fields are optional, and repeated fields are required"?

Reply:
I meant the second one.  I changed "single" to "optional", as in Protocol Buffers, to try to make it clearer.